### ISSUE TITLE
Create SourceTrail Github organization

### ISSUE THREAD

**Issue by fire** [CoatiSoftware/Sourcetrail#1225]
Instead of archiving CoatiSoftware /
Sourcetrail Github organization. I propose making a SourceTrail Github organization.

Thoughts and comments?

**Comment by mlangkabel** [comment]
What would be the advantage of that? How would it be useful?

**Comment by fire** [comment]
The stated plan is for this set of repos to be archived. So the first thing would be to create a mirror and an new organization.

**Comment by frhun** [comment]
I was in the middle of packaging sourcetrail for flatpak, to put it on flathub, and thus make it discoverable on several linux distributions. (flatpak also has the advantage that flatpak packages can go a very long time without being maintained, because they are self contained for their dependencies (in a way more reliable and portable way than AppImages ever could BTW).
It also eliminates most of the distribution nightmare that Linux usually is.
I stopped partly because didn't want to refer to a dead repo.
Flatpaks are very reliant on namespaces, and thus it would be bad to create it now, and change it to a non com.sourcetrail namespace later, if and when a fork and maintainer comes along.

Long story short.. i would like for someone (even if it's not someone in a maintainer position) to take stewardship of the website and namespace, who could then give that to an eventual new project maintainer if a suitable fork comes along.

**Comment by LouisStAmour** [comment]
I’d be happy to join a SourceTrail organization on GitHub were one to be established. 

I’ve had an idea in the back of my head to rewrite the Qt framework frontend in React and TypeScript both so that there could be one central store for state instead of updating the state of each “pane” separately. (E.g. the search submit or click of a updates each pane manually right now…) 

But also because I think a web-based frontend might integrate better into IDEs and make it a bit easier to get new contributions and new language features added graphically to the display.

It would be ideal if we could continue to use the SQLite database/project format for such a frontend so the backend code can be shared as needed. The hardest part is probably rendering the interactive graph - doable, just a lot of work as was the Qt version to begin with, obviously.

And since it’s a copy of an existing project that was previously maintained/commercial, I’d hesitated to get started on a rewrite. But more and more I’m thinking I might pick it up and try rewriting the graph display/interaction engine in TypeScript/React, maybe, perhaps for now backed by SQLite compiled via WebASM or something to just get the data flowing…

**Comment by frhun** [comment]
Is there a trademark on sourcetrail?

**Comment by RegisPerdreau** [comment]
I want to take part. For the time being, i need to learn and may be write some thoughts about how to start contribution. This software is useful for large project, i have tested it in LibreOffice context. We have to "make hay while the sun shines".

**Comment by RegisPerdreau** [comment]
We have to find a name, fork the repo, give access, define a new website. May be we can

> Is there a trademark on sourcetrail?

Yes, author said he keeps the name

**Comment by simonsan** [comment]
I want to let here some links:

- https://linuxfoundation.org/projects/hosting/ or opencollective.com/ for building a sustainable community
- liberapay.com instead of a Patreon
- libera.chat/ or zulipchat.com/ for communication
- maybe a rewrite of the C++ and Java `core` in Rust with `egui` or `iced` or `tauri` GUI framework to let go the burden of cross-platform (can be even compiled to WASM and used in the browser, huge opportunities e.g. integrating with GitHub, as a Plugin to VSCode/VSCodium etc.)

**Comment by simonsan** [comment]
> I’d be happy to join a SourceTrail organization on GitHub were one to be established.
> 
> I’ve had an idea in the back of my head to rewrite the Qt framework frontend in React and TypeScript partly so that there could be one central store for state instead of updating the state of each “pane” separately. (E.g. the search submit or click of a updates each pane manually right now since each object in Qt has its own state in addition to a render method)
> 
> But also because I think a web-based frontend might integrate better into IDEs and make it a bit easier to get new contributions and new language features added graphically to the display.
> 
> It would be ideal if we could continue to use the SQLite database/project format for such a frontend so the backend code can be shared as needed. The hardest part is probably rendering the interactive graph - doable, just a lot of work as was the Qt version to begin with, obviously.
> 
> And since it’s a copy of an existing project that was previously maintained/commercial, I’d hesitated to get started on a rewrite. But more and more I’m thinking I might pick it up and try rewriting the graph display/interaction engine in TypeScript/React, for now backed by SQLite compiled via WebASM or something to just get the data flowing…

Sounds reasonable, indeed. Recommend to take a look at https://tauri.studio/ (FLOSS Electron alternative) and https://deno.land/ if you want to stay in pure TypeScript.

**Comment by egraether** [comment]
We are still planning to archive this repository before the end of the year. This will probably happen on December 14th.

You may fork the repository and continue working on the project as you wish, the legal conditions of the software license apply. However, we would like to ask you to use a different name in order to prevent confusion of the users.
